PyPDF2/_writer.py

```diff
@@ -622,6 +622,8 @@ def update_page_form_field_values(
         """
         self.set_need_appearances_writer()
         # Iterate through pages, update field values
+        if PG.ANNOTS not in page:
+            return
```

Comment:
Shouldn't we add the attribute if it doesn't exist, e.g.:

```suggestion
        if PG.ANNOTS not in page:
            page[PG.ANNOTS] = DictionaryObject()
```

Reply:
if the form has no fields to field, adding the PG.ANNOTS field may be of no use and quite confusing after (looking at the page structure you may think after there is so fields to fill up whereas there isn't)

Reply:
> if the form has no fields to field

I guess there was a typo? I don't understand this.

At the very least we should emit a warning ... if we are in strict mode maybe even an exception?

Reply:
if the form has no fields to _be filled_, adding the PG.ANNOTS field may be of no use and quite confusing after (looking at the page structure you may think after there is so fields to fill up whereas there isn't) ; more consistent with PDF philosophy : if you don't need it, it is not in the dictionary (no fields means logically no PG.ANNOTS entry)

Reply:
I understand why you don't want to add it. It makes sense to me now.

However, looking at it from the users perspective: The user wants to fill a form field that doesn't exist. PyPDF2 executes sucessfully without an exception, without a warning, without a log message. No field is filled. That feels wrong to me.

A warning log message - something like "Filling fields was skipped as the page does not have any annotations" would give the user a chance to fix the code.